src/array/list/mutable.rs

```diff
@@ -142,6 +142,23 @@ impl<O: Offset, M: MutableArray> MutableListArray<O, M> {
         Self::new_from(values, data_type, capacity)
     }
 
+    /// Creates a new [`MutableListArray`] from a [`MutableArray`], [`Offsets`] and
+    /// [`MutableBitmap`].
+    pub fn new_from_mutable(
+        values: M,
+        offsets: Offsets<O>,
+        validity: Option<MutableBitmap>,
+    ) -> Self {
+        assert_eq!(values.len(), offsets.last().to_usize());
+        let data_type = values.data_type().clone();
```

Comment:
@sundy-li, thanks for the quick response. Unfortunately I missed on the dtype -- it should be the following:
```rust
        let data_type = ListArray::<O>::default_datatype(values.data_type().clone());
```